File: python/copass-management/src/copass_management/tools/update_agent_prompt.py
from __future__ import annotations

from typing import Any, Dict, TYPE_CHECKING

if TYPE_CHECKING:
    from copass_management.registrar import ToolContext


async def update_agent_prompt(ctx: "ToolContext", input: Dict[str, Any]) -> Any:
    slug = str(input["slug"])
    agent = await ctx.client.agents.update(
        ctx.sandbox_id,
        slug,
        system_prompt=str(input["system_prompt"]),
    )
    return {"agent": agent}

File: typescript/packages/management/src/tools/update_agent_prompt.ts
import type { ToolContext, ToolHandler } from '../registrar.js';

export const updateAgentPrompt: ToolHandler = async (
  ctx: ToolContext,
  input: Record<string, unknown>,
) => {
  const slug = String(input.slug);
  const agent = await ctx.client.agents.update(ctx.sandboxId, slug, {
    system_prompt: String(input.system_prompt),
  });
  return { agent };
};
